### PR TITLE
[tests/MSBuildDeviceIntegration] Add test for dll's which have the same size.

### PR DESCRIPTION
**Pull request by dellis1972**
Context https://github.com/xamarin/monodroid/issues/1123

There seems to be a bug in the new fast deployment system
where if the dll was modified BUT had exactly the same size
as the existing dll on the device it would not be uploade.

This commit adds a test to make sure this does work as
expected.

### CODE REVIEW

**Review by radekdoulik — COMMENTED** [review]
For peace of mind I would add one more `Assert` with check that the 2 assembly files are indeed the same size?

File: tests/MSBuildDeviceIntegration/Tests/InstallTests.cs [hunk 1]
```diff
@@ -504,15 +506,22 @@ public void IncrementalFastDeployment ()
...
+				Assert.IsTrue (logLines.Any (l => l.Contains ("NotifySync CopyFile") && l.Contains ("UnnamedProject.dll")), "UnnamedProject.dll should have been uploaded");
+				Assert.IsTrue (logLines.Any (l => l.Contains ("NotifySync CopyFile") && l.Contains ("Library1.dll")), "Library1.dll should have been uploaded");
+				Assert.IsTrue (logLines.Any (l => l.Contains ("NotifySync CopyFile") && l.Contains ("Library2.dll")), "Library2.dll should have been uploaded");
 				var firstInstallTime = builder.LastBuildTime;
+				builder.BuildLogFile = "install2.log";
 				Assert.IsTrue (builder.Install (app, doNotCleanupOnUpdate: true, saveProject: false), "Second install should have succeeded.");
 				var secondInstallTime = builder.LastBuildTime;
 
 				var filesToTouch = new [] {
-					Path.Combine (rootPath, lib1.ProjectName, "Class1.cs"),
```
Comment (jonathanpeppers) [inline tests/MSBuildDeviceIntegration/Tests/InstallTests.cs:515]: Maybe I'm missing something, but does `Class1.cs` not ever get `Touch()` called on it?

Comment (dellis1972) [reply tests/MSBuildDeviceIntegration/Tests/InstallTests.cs:515]: We change the actual TextContent here https://github.com/xamarin/xamarin-android/pull/5229/files#diff-34b4814429c211ae0bf929ddaa36bbc4eb8c13de248a744fa75e7a8f333fb9aeR533

Comment (jonathanpeppers) [reply tests/MSBuildDeviceIntegration/Tests/InstallTests.cs:515]: I think the `TimeStamp` has to change for it to save the `.cs` file?

Comment (dellis1972) [reply tests/MSBuildDeviceIntegration/Tests/InstallTests.cs:515]: I'll test that out

Comment (dellis1972) [reply tests/MSBuildDeviceIntegration/Tests/InstallTests.cs:524]: The timestamp is set by this line https://github.com/xamarin/xamarin-android/pull/5229/files#diff-34b4814429c211ae0bf929ddaa36bbc4eb8c13de248a744fa75e7a8f333fb9aeR539

**Review by jonathanpeppers — APPROVED** [review]

**Pull request merged**